oracle/src/confirmRelay.js

```diff
@@ -0,0 +1,155 @@
+const processAMBCollectedSignatures = require('./events/processAMBCollectedSignatures')(config)
+const processAMBAffirmationRequests = require('./events/processAMBAffirmationRequests')(config)
+
+const { getTokensState } = require('./utils/tokenState')
+
+const web3Instance = config.web3
+const bridgeContract = new web3Instance.eth.Contract(config.bridgeAbi, config.bridgeContractAddress)
+let { eventContractAddress } = config
+let eventContract = new web3Instance.eth.Contract(config.eventAbi, eventContractAddress)
+
+async function initialize() {
+  try {
+    const checkHttps = checkHTTPS(process.env.ORACLE_ALLOW_HTTP_FOR_RPC, logger)
+
+    rpcUrlsManager.homeUrls.forEach(checkHttps('home'))
+    rpcUrlsManager.foreignUrls.forEach(checkHttps('foreign'))
+
+    connectWatcherToQueue({
+      queueName: config.queue,
+      workerQueue: config.workerQueue,
+      cb: runMain
+    })
+  } catch (e) {
+    logger.error(e)
+    process.exit(EXIT_CODES.GENERAL_ERROR)
+  }
+}
+
+async function runMain({ sendToQueue, sendToWorker }) {
+  try {
+    if (connection.isConnected()) {
+      if (config.maxProcessingTime) {
+        await watchdog(() => main({ sendToQueue, sendToWorker, txHash }), config.maxProcessingTime, () => {
+          logger.fatal('Max processing time reached')
+          process.exit(EXIT_CODES.MAX_TIME_REACHED)
+        })
+      } else {
+        await main({ sendToQueue, sendToWorker, txHash })
+      }
+    } else {
+      setTimeout(() => {
+        runMain({ sendToQueue, sendToWorker })
+      }, config.pollingInterval)
+    }
+  } catch (e) {
+    logger.error(e)
+  }
+}
+
+function processEvents(events) {
+  switch (config.id) {
+    case 'native-erc-signature-request':
+    case 'erc-erc-signature-request':
+    case 'erc-native-signature-request':
+      return processSignatureRequests(events)
+    case 'native-erc-collected-signatures':
+    case 'erc-erc-collected-signatures':
+    case 'erc-native-collected-signatures':
+      return processCollectedSignatures(events)
+    case 'native-erc-affirmation-request':
+    case 'erc677-erc677-affirmation-request':
+    case 'erc-native-affirmation-request':
+    case 'erc-erc-affirmation-request':
+      return processAffirmationRequests(events)
+    case 'erc-erc-transfer':
+    case 'erc-native-transfer':
+      return processTransfers(events)
+    case 'amb-signature-request':
+      return processAMBSignatureRequests(events)
+    case 'amb-collected-signatures':
+      return processAMBCollectedSignatures(events)
+    case 'amb-affirmation-request':
+      return processAMBAffirmationRequests(events)
+    default:
+      return []
+  }
+}
+
+async function checkConditions() {
+  let state
+  switch (config.id) {
+    case 'erc-native-transfer':
+      logger.debug('Getting token address to listen Transfer events')
+      state = await getTokensState(bridgeContract, logger)
+      updateEventContract(state.bridgeableTokenAddress)
```

Comment:
Do we need this? As far as I recall it was used to handle a case when `bridgeableTokenAddress` is not defined yet. Please double check if we can simplify the code here.

Reply:
Yes, you are right, it will work fine without this. I simplified it in b4fedc3.